byzcoin/trie/node.go

```diff
@@ -0,0 +1,165 @@
+package trie
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"hash"
+
+	"github.com/dedis/protobuf"
+)
+
+type nodeType int
+
+const (
+	typeInterior nodeType = iota
+	typeEmpty
+	typeLeaf
+)
+
+func (ty nodeType) toBytes() []byte {
+	switch ty {
+	case typeInterior:
+		return []byte{0}
+	case typeEmpty:
+		return []byte{1}
+	case typeLeaf:
+		return []byte{2}
+	default:
+		panic("no such type")
+	}
+}
+
+type interiorNode struct {
+	Left  []byte
+	Right []byte
+}
+
+func (n *interiorNode) hash() []byte {
+	h := sha256.New()
```

Comment:
Why does this hash not include the `typeInteriorNode`?

Reply:
Same question for `leafNode`

Reply:
actually, there is an implementation error, will be fixed as a part of - https://github.com/dedis/cothority/pull/1534#discussion_r227795381